### ISSUE TITLE
Lock Orientation

### ISSUE THREAD

**Issue by rslkr** [yoshisuga/MAME4iOS#80]
Hello, Can you add an option 'Screen Lock Orientation' please?Sometimes I want to keep it Landscape mode. iOS control center Lock Orientation only works for Portrait mode!
One more thing about Netplay. One of my friend who lives overseas and he has this app. begging me to play over internet. We've tried several method, didn't work out! Can you add this option as well please?
Also I was wondering, are you going to update for recent MAME version?
And lastly I realized that I've been using this app for a long time.
So I'm willing to donate with Paypal if you like.
Thanks.

**Comment by yoshisuga** [comment]
Hi! i can look into adding an option to lock the current screen orientation, sure.

With regards to netplay, i'm not sure if it still works - it's something i've yet to really test, but definitely worth looking into when I have the time.

Regarding the supporting the latest MAME version, yes I do plan to update it at some point, but it will take a lot of time to rework it into the app. It's not a matter of simply recompiling, but also rebuilding the app to work with the updated "OSD" api.

I was able to figure out how the recent versions of MAME get built, and was able to get the latest version of MAME (0.212) to compile for iOS and tvOS as a libretro core for RetroArch.  So at least some progress has been made, and we'll eventually have an updated version of MAME4iOS.

I also contribute to RetroArch for iOS/tvOS, so if you want to run MAME games for versions later than 0.139, try the MAME (0.212) or MAME2015 (0.165?) cores for RetroArch here:
https://buildbot.libretro.com/stable/1.7.8/apple/ios/RetroArch.ipa

I do like having MAME be a standalone app, and I do plan on supporting it for as long as I can.

I've found that the difference in the amount of games supported between 0.212 and 0.139 was not as much as I thought. There are some cool new additions that run well, like Star Wars Arcade for Sega Model 1, Virtua Fighter 2 for Sega Model 2. But MAME4iOS still runs a great number of arcade titles with pretty good speed.

Thanks for enjoying the app, and no need for a donation! :)

**Comment by rslkr** [comment]
Thanks for reply. I really do enjoying this app. This was one of my dream, whole arcade games in my pocket. Because of you this app. still alive and available, really appreciated.

Peer to Peer Netplay works perfectly fine and the app. itself of course.

I don't know who wrote the codes, David Valdeita (Seleuco) himself or MAME Team helped him. I found  //DAV HACK codes in some ''.c'' files. And in 139 there is only ''src'' folder, add to ''makefile'' Recent MAME has more folders in it. This is the hard part I guess.

I'm not a developer or programmer but I figured out how to add unofficial rom into MAME.
MAME only checks rom's CRC and SHA1. I edited these three files ''cps1.c(video) cps1.c(driver) mamedriv.c'' rom name: sf2jh  Street Fighter II: The World Warrior (Yumeji Hack SF2++)

But I'm not good at Xcode I can simply fallow instruction.
I don't know how you add the opacity slider that I asked before. I tried to fix it but its beyond my knowledge. Could you check my last comment:[https://github.com/yoshisuga/MAME4iOS/issues/27](url)
Could you add something like this as well: [https://www.cocoacontrols.com/controls/rgbcolorslider](url) rgb slider for buttons and joystick to change colors please? This is my last request I won't bother you anymore I promise.

I tested all exist emulators Happychick, GBA, Provenance etc. None of them good as like this app. not even close, in my opinion.
I know you're doing hard work on RetroArch but the app needs major update. The UI is terrible not user friendly and complicated. Its just not for me sorry.

Anyway thanks for your effort and time.

**Comment by mrjschulte** [comment]
Hi, can you try the net play improvements we've made to the new merge as of today and let us know if this works?

**Comment by mrjschulte** [comment]
Looks like using Orientation lock at the OS Level works, so closing this ticket out if we don't hear back otherwise but feel free to open a new issue for each specific issue to help w tracking if they still need to exist.

**Comment by rslkr** [comment]
Hi, @mrjschulte 

> Hi, can you try the net play improvements we've made to the new merge as of today and let us know if this works?

I had to update the XCode to run the latest M4i. (ios 14)
Tested netplay, its ok. Like @ToddLa point it out it's not working via Bluetooth.

But my request is about netplay to play over internet (of course over WiFi) like using the **kaillera server** or similar.
I did little research, MAME core not built-in netplay it requires server... add to the core.

Also Orientation lock still works for Portrait mode not on the Landscape!

I should've opened the issues one by one, not all of them here...
Anyway I will close this issue, but If you are really allow me to open another one(s)?

@ToddLa thanks for the position of the HUD 👍
